pkg/secrets/aws_secrets_manager_test.go

```diff
@@ -0,0 +1,207 @@
+package secrets
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"io/ioutil"
+	"testing"
+)
+
+type MockAwsSecretsManagerClient struct {
+	mock.Mock
+	payload string
+}
+
+func (m *MockAwsSecretsManagerClient) FetchSecret(secretName string) (*secretsmanager.GetSecretValueOutput, error) {
+	mockPayloadBytes, _ := ioutil.ReadFile("../../test/aws-secrets-manager/" + m.payload)
+	res := &secretsmanager.GetSecretValueOutput{}
+	err := json.Unmarshal(mockPayloadBytes, res)
+	if err != nil {
+		fmt.Println(err.Error())
+		panic(err)
+	}
```

Comment:
I'm not exactly sure how you'd refactor this but you typically shouldn't `panic` in tests. This is more common in helper functions but usually you'd pass the test's `testing.T` into the mock and `t.Fatal` on unexpected errors as it's generally more readable/won't fail the entire test suite. 

https://speakerdeck.com/mitchellh/advanced-testing-with-go?slide=27

One possible solution - make a test helper that will perform this file read/unmarshaling and you can set on the mock.

```
type fixtureReader func(filePath string) (*secretsManager.GetSecretValueOutput{}, error)

func fixtureReaderHelper(t *testing.T) fixtureReader {
	return func(filePath string) (*secretsManager.GetSecretValueOutput{}, error) {
		//read a file
		if err := json.Unmarshal(fileContents, res); err != nil {
			// calls to fatal will stop the test for you
			t.Fatalf("could not unmarshal fixture JSON", e)
		}
		// return your stuff.
		
	}
}
```

Then, add a `fixtureReader` to the mock that you generate when you create your test cases. 

There's probably a cleaner/better way. That's just what came to mind off the top of my head.

Reply:
![image](https://user-images.githubusercontent.com/711726/74965987-18565900-53cb-11ea-9a60-f61649ae2df8.png)

![image](https://user-images.githubusercontent.com/711726/74966014-22785780-53cb-11ea-9a4d-dd46de41cf78.png)

☝️ I think that will work.

Reply:
![image](https://user-images.githubusercontent.com/711726/74966224-89960c00-53cb-11ea-8c70-957b7d6a2d05.png)